### PR TITLE
Upgrade to YARP v0.8.0 (indexer changes)

### PR DESCRIPTION
**Pull request by andyw8**
### Motivation

Closes https://github.com/Shopify/ruby-lsp/issues/912

### Implementation

Handle the new `OrWriteNode` in YARP v0.8.0.

### Automated Tests

Updated

### Manual Tests

n/a

### CODE REVIEW

File: lib/ruby_indexer/test/constant_test.rb [hunk 1]
```diff
@@ -27,8 +27,8 @@ class ::Bar
         end
       RUBY
 
-      assert_entry("FOO", Index::Entry::Constant, "/fake/path/foo.rb:0-0:0-8")
```
Comment (andyw8) [inline lib/ruby_indexer/test/constant_test.rb:30]: Previous we were referencing the whole assignment, but I think it makes more sense to only reference the constant name?

**Review by vinistock — APPROVED** [review]
Looks good. Let's just see what's happening with Ruby head

**Comment by andyw8** [comment]
>  I believe the current implementation isn't handling &&= and some other kinds of operator assignments, should we handle that here or separately?

Actually `&&=` is not valid for constants, but I wonder if anything similar is...

**Comment by vinistock** [comment]
I think we can focus this PR on fixing the breaking changes, but we should indeed consider these other assignment operators as other entries for the same constant.

**Comment by andyw8** [comment]
> I think we can focus this PR on fixing the breaking changes, but we should indeed consider these other assignment operators as other entries for the same constant.

https://github.com/Shopify/ruby-lsp/issues/916

**Pull request merged**